packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -1521,4 +1527,104 @@ void main() {
     expect((mergeableMaterial.children.first as MaterialSlice).color, firstPanelColor);
     expect((mergeableMaterial.children.last as MaterialSlice).color, secondPanelColor);
   });
+
+  testWidgets('ExpansionPanelList expanded color test', (WidgetTester tester) async {
+    const Color expandedColor = Colors.orange;
+    const Color color = Colors.red;
+
+    const Key firstPanelKey = Key('firstPanelKey');
+    const Key secondPanelKey = Key('secondPanelKey');
+
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: SingleChildScrollView(
+          child: SimpleExpansionPanelListTestWidget(
+            firstPanelKey: firstPanelKey,
+            secondPanelKey: secondPanelKey,
+            color: color,
+            expandedColor: expandedColor,
+          ),
+        ),
+      ),
+    );
+
+    final Finder finder = find.descendant(
+      of: find.ancestor(
+        of: find.byKey(firstPanelKey),
+        matching: find.byType(Row),
+      ),
+      matching: find.byType(IconButton),
+    );
+
+    IconButton iconButton = tester.widget(finder);
+
+    expect(iconButton.color, color);
+
+    await tester.tap(find.byType(ExpandIcon).at(0));
+    await tester.pumpAndSettle();
+
+    iconButton = tester.widget(find.byType(IconButton).at(0));
+
+    expect(iconButton.color, expandedColor);
+  });
+
+  testWidgets('ExpansionPanelRadio expanded color test', (WidgetTester tester) async {
+    const Color expandedColor = Colors.orange;
+    const Color color = Colors.red;
+
+    const Key firstPanelKey = Key('firstPanelKey');
+    const Key secondPanelKey = Key('secondPanelKey');
```

Comment:
Since we can find the IconButton without needing the key (above conversation), you can remove them.